pkg/artifactory/resource_artifactory_saml_settings.go

```diff
@@ -0,0 +1,203 @@
+	EmailAttribute            string `yaml:"emailAttribute" json:"emailAttribute"`
+	GroupAttribute            string `yaml:"groupAttribute" json:"groupAttribute"`
+	LoginUrl                  string `yaml:"loginUrl" json:"loginUrl"`
+	LogoutUrl                 string `yaml:"logoutUrl" json:"logoutUrl"`
+	NoAutoUserCreation        bool   `yaml:"noAutoUserCreation" json:"noAutoUserCreation"`
+	ServiceProviderName       string `yaml:"serviceProviderName" json:"serviceProviderName"`
+	AllowUserToAccessProfile  bool   `yaml:"allowUserToAccessProfile" json:"allowUserToAccessProfile"`
+	AutoRedirect              bool   `yaml:"autoRedirect" json:"autoRedirect"`
+	SyncGroups                bool   `yaml:"syncGroups" json:"syncGroups"`
+	VerifyAudienceRestriction bool   `yaml:"verifyAudienceRestriction" json:"verifyAudienceRestriction"`
+}
+
+func resourceArtifactorySamlSettings() *schema.Resource {
+	return &schema.Resource{
+		Update: resourceSamlSettingsUpdate,
+		Create: resourceSamlSettingsUpdate,
+		Delete: resourceSamlSettingsDelete,
+		Read:   resourceSamlSettingsRead,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"enable": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"certificate": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"email_attribute": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"group_attribute": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"login_url": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"logout_url": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"no_auto_user_creation": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"service_provider_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"allow_user_to_access_profile": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"auto_redirect": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"sync_groups": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"verify_audience_restriction": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+		},
+	}
+}
+
+func resourceSamlSettingsRead(d *schema.ResourceData, m interface{}) error {
+	c := m.(*ArtClient).ArtNew
+	serviceDetails := c.GetConfig().GetServiceDetails()
+	httpClientDetails := serviceDetails.CreateHttpClientDetails()
+
+	samlSettings := SamlSettings{}
+
+	_, body, _, err := c.Client().SendGet(fmt.Sprintf("%sapi/saml/config", serviceDetails.GetUrl()), false, &httpClientDetails)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve data from <base_url>/artifactory/api/saml/config during Read")
+	}
+
+	err = json.Unmarshal(body, &samlSettings)
```

Comment:
Ok, I am befuddled: You `GET` and it's `JSON`, but below, you `PATCH` and send it `YAML`?

Reply:
`GET`
 The three undocumented API endpoints return JSON,
`<base-url>/artifactory/api/securityconfig`
`<base-url>/artifactory/api/oauth`
`<base-url>/artifactory/api/saml/config`

`PATCH` 
The documented endpoint for configuration only supports a YAML media type,
`<base-url>/artifactory/api/system/configuration`
[JFrog Artifactory YAML Configuration Documentation](https://www.jfrog.com/confluence/display/JFROG/Artifactory+YAML+Configuration#ArtifactoryYAMLConfiguration-Security(Generalsecurity,PasswordPolicy,LDAP,SAML,OAuth,HTTPSSO,Crowd))